### PR TITLE
Include Normalize.css if Bootstrap is not present

### LINKED ISSUE #242: vendor.css file error - Failed to load resource: the server responded with a status of 404 (Not Found)
Created new project using yeoman webapp generator.

After running `grunt build` and `grunt serve:dist`  i am getting vendor.css file not found error.

vendorcss

When i checked all the folders i could only find it in app/.tmp/concat/styles/ folder, whereas main.css file is located at app/.tmp/concat/styles/ and .tmp/styles/

When i saw the `grunt build` out put i found following lines which might be related to this:

```
concat:
  { generated: 
   { files: 
      [ { dest: '.tmp/concat/styles/vendor.css', src: [] },
        { dest: '.tmp/concat/styles/main.css',
          src: [ '.tmp/styles/main.css' ] },
```

**Out put of : yo --version && echo $PATH $NODE_PATH && node -e 'console.log(process.platform, process.versions)' && cat Gruntfile.js**

```
    $ yo --version && echo $PATH $NODE_PATH && node -e 'console.log(process.platform, process.versions)' && cat Gruntfile.js
    1.1.2
    /home/auto-w24/.rvm/gems/ruby-2.0.0-p353/bin:/home/auto-w24/.rvm/gems/ruby-2.0.0-p353@global/bin:/home/auto-w24/.rvm/rubies/ruby-2.0.0-p353/bin:/home/auto-w24/.rvm/bin:/home/auto-w24/local/bin:/home/auto-w24/local/bin:/home/auto-w24/local/bin:/home/auto-w24/local/bin:/home/auto-w24/.local/bin:/home/auto-w24/local/bin:/home/auto-w24/local/bin:/home/auto-w24/local/bin:/usr/lib/lightdm/lightdm:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/share/ruby-rvm/bin /usr/lib/nodejs:/usr/lib/node_modules:/usr/share/javascript:/home/auto-w24/.nvm/v0.10.24/lib/node_modules:/usr/local/lib/node_modules
    linux { http_parser: '1.0',
      node: '0.10.24',
      v8: '3.14.5.9',
      ares: '1.9.0-DEV',
      uv: '0.10.21',
      zlib: '1.2.3',
      modules: '11',
      openssl: '1.0.1e' }
    // Generated on 2014-01-09 using generator-webapp 0.4.6
    'use strict';
    // # Globbing
    // for performance reasons we're only matching one level down:
    // 'test/spec/{,*/}*.js'
    // use this if you want to recursively match all subfolders:
    // 'test/spec/**/*.js'

    module.exports = function (grunt) {

        // Load grunt tasks automatically
        require('load-grunt-tasks')(grunt);

        // Time how long tasks take. Can help when optimizing build times
        require('time-grunt')(grunt);

        // Define the configuration for all the tasks
        grunt.initConfig({

            // Project settings
            yeoman: {
                // Configurable paths
                app: 'app',
                dist: 'dist'
            },

            // Watches files for changes and runs tasks based on the changed files
            watch: {
                js: {
                    files: ['<%= yeoman.app %>/scripts/{,*/}*.js'],
                    tasks: ['jshint'],
                    options: {
                        livereload: true
                    }
                },
                jstest: {
                    files: ['test/spec/{,*/}*.js'],
                    tasks: ['test:watch']
                },
                gruntfile: {
                    files: ['Gruntfile.js']
                },
                compass: {
                    files: ['<%= yeoman.app %>/styles/{,*/}*.{scss,sass}'],
                    tasks: ['compass:server', 'autoprefixer']
                },
                styles: {
                    files: ['<%= yeoman.app %>/styles/{,*/}*.css'],
                    tasks: ['newer:copy:styles', 'autoprefixer']
                },
                livereload: {
                    options: {
                        livereload: '<%= connect.options.livereload %>'
                    },
                    files: [
                        '<%= yeoman.app %>/{,*/}*.html',
                        '.tmp/styles/{,*/}*.css',
                        '<%= yeoman.app %>/images/{,*/}*.{gif,jpeg,jpg,png,svg,webp}'
                    ]
                }
            },

            // The actual grunt server settings
            connect: {
                options: {
                    port: 9000,
                    livereload: 35729,
                    // Change this to '0.0.0.0' to access the server from outside
                    hostname: 'localhost'
                },
                livereload: {
                    options: {
                        open: true,
                        base: [
                            '.tmp',
                            '<%= yeoman.app %>'
                        ]
                    }
                },
                test: {
                    options: {
                        port: 9001,
                        base: [
                            '.tmp',
                            'test',
                            '<%= yeoman.app %>'
                        ]
                    }
                },
                dist: {
                    options: {
                        open: true,
                        base: '<%= yeoman.dist %>',
                        livereload: false
                    }
                }
            },

            // Empties folders to start fresh
            clean: {
                dist: {
                    files: [{
                        dot: true,
                        src: [
                            '.tmp',
                            '<%= yeoman.dist %>/*',
                            '!<%= yeoman.dist %>/.git*'
                        ]
                    }]
                },
                server: '.tmp'
            },

            // Make sure code styles are up to par and there are no obvious mistakes
            jshint: {
                options: {
                    jshintrc: '.jshintrc',
                    reporter: require('jshint-stylish')
                },
                all: [
                    'Gruntfile.js',
                    '<%= yeoman.app %>/scripts/{,*/}*.js',
                    '!<%= yeoman.app %>/scripts/vendor/*',
                    'test/spec/{,*/}*.js'
                ]
            },


            // Mocha testing framework configuration options
            mocha: {
                all: {
                    options: {
                        run: true,
                        urls: ['http://<%= connect.test.options.hostname %>:<%= connect.test.options.port %>/index.html']
                    }
                }
            },



            // Compiles Sass to CSS and generates necessary files if requested
            compass: {
                options: {
                    sassDir: '<%= yeoman.app %>/styles',
                    cssDir: '.tmp/styles',
                    generatedImagesDir: '.tmp/images/generated',
                    imagesDir: '<%= yeoman.app %>/images',
                    javascriptsDir: '<%= yeoman.app %>/scripts',
                    fontsDir: '<%= yeoman.app %>/styles/fonts',
                    importPath: '<%= yeoman.app %>/bower_components',
                    httpImagesPath: '/images',
                    httpGeneratedImagesPath: '/images/generated',
                    httpFontsPath: '/styles/fonts',
                    relativeAssets: false,
                    assetCacheBuster: false
                },
                dist: {
                    options: {
                        generatedImagesDir: '<%= yeoman.dist %>/images/generated'
                    }
                },
                server: {
                    options: {
                        debugInfo: true
                    }
                }
            },

            // Add vendor prefixed styles
            autoprefixer: {
                options: {
                    browsers: ['last 1 version']
                },
                dist: {
                    files: [{
                        expand: true,
                        cwd: '.tmp/styles/',
                        src: '{,*/}*.css',
                        dest: '.tmp/styles/'
                    }]
                }
            },

            // Automatically inject Bower components into the HTML file
            'bower-install': {
                app: {
                    html: '<%= yeoman.app %>/index.html',
                    ignorePath: '<%= yeoman.app %>/'
                }
            },

            // Renames files for browser caching purposes
            rev: {
                dist: {
                    files: {
                        src: [
                            '<%= yeoman.dist %>/scripts/{,*/}*.js',
                            '<%= yeoman.dist %>/styles/{,*/}*.css',
                            '<%= yeoman.dist %>/images/{,*/}*.{gif,jpeg,jpg,png,webp}',
                            '<%= yeoman.dist %>/styles/fonts/{,*/}*.*'
                        ]
                    }
                }
            },

            // Reads HTML for usemin blocks to enable smart builds that automatically
            // concat, minify and revision files. Creates configurations in memory so
            // additional tasks can operate on them
            useminPrepare: {
                options: {
                    dest: '<%= yeoman.dist %>'
                },
                html: '<%= yeoman.app %>/index.html'
            },

            // Performs rewrites based on rev and the useminPrepare configuration
            usemin: {
                options: {
                    assetsDirs: ['<%= yeoman.dist %>']
                },
                html: ['<%= yeoman.dist %>/{,*/}*.html'],
                css: ['<%= yeoman.dist %>/styles/{,*/}*.css']
            },

            // The following *-min tasks produce minified files in the dist folder
            imagemin: {
                dist: {
                    files: [{
                        expand: true,
                        cwd: '<%= yeoman.app %>/images',
                        src: '{,*/}*.{gif,jpeg,jpg,png}',
                        dest: '<%= yeoman.dist %>/images'
                    }]
                }
            },
            svgmin: {
                dist: {
                    files: [{
                        expand: true,
                        cwd: '<%= yeoman.app %>/images',
                        src: '{,*/}*.svg',
                        dest: '<%= yeoman.dist %>/images'
                    }]
                }
            },
            htmlmin: {
                dist: {
                    options: {
                        collapseBooleanAttributes: true,
                        collapseWhitespace: true,
                        removeAttributeQuotes: true,
                        removeCommentsFromCDATA: true,
                        removeEmptyAttributes: true,
                        removeOptionalTags: true,
                        removeRedundantAttributes: true,
                        useShortDoctype: true
                    },
                    files: [{
                        expand: true,
                        cwd: '<%= yeoman.dist %>',
                        src: '{,*/}*.html',
                        dest: '<%= yeoman.dist %>'
                    }]
                }
            },

            // By default, your `index.html`'s <!-- Usemin block --> will take care of
            // minification. These next options are pre-configured if you do not wish
            // to use the Usemin blocks.
            // cssmin: {
            //     dist: {
            //         files: {
            //             '<%= yeoman.dist %>/styles/main.css': [
            //                 '.tmp/styles/{,*/}*.css',
            //                 '<%= yeoman.app %>/styles/{,*/}*.css'
            //             ]
            //         }
            //     }
            // },
            // uglify: {
            //     dist: {
            //         files: {
            //             '<%= yeoman.dist %>/scripts/scripts.js': [
            //                 '<%= yeoman.dist %>/scripts/scripts.js'
            //             ]
            //         }
            //     }
            // },
            // concat: {
            //     dist: {}
            // },

            // Copies remaining files to places other tasks can use
            copy: {
                dist: {
                    files: [{
                        expand: true,
                        dot: true,
                        cwd: '<%= yeoman.app %>',
                        dest: '<%= yeoman.dist %>',
                        src: [
                            '*.{ico,png,txt}',
                            '.htaccess',
                            'images/{,*/}*.webp',
                            '{,*/}*.html',
                            'styles/fonts/{,*/}*.*',
                            'bower_components/sass-bootstrap/fonts/*.*'
                        ]
                    }]
                },
                styles: {
                    expand: true,
                    dot: true,
                    cwd: '<%= yeoman.app %>/styles',
                    dest: '.tmp/styles/',
                    src: '{,*/}*.css'
                }
            },


            // Generates a custom Modernizr build that includes only the tests you
            // reference in your app
            modernizr: {
                devFile: '<%= yeoman.app %>/bower_components/modernizr/modernizr.js',
                outputFile: '<%= yeoman.dist %>/bower_components/modernizr/modernizr.js',
                files: [
                    '<%= yeoman.dist %>/scripts/{,*/}*.js',
                    '<%= yeoman.dist %>/styles/{,*/}*.css',
                    '!<%= yeoman.dist %>/scripts/vendor/*'
                ],
                uglify: true
            },

            // Run some tasks in parallel to speed up build process
            concurrent: {
                server: [
                    'compass:server',
                    'copy:styles'
                ],
                test: [
                    'copy:styles'
                ],
                dist: [
                    'compass',
                    'copy:styles',
                    'imagemin',
                    'svgmin'
                ]
            }
        });


        grunt.registerTask('serve', function (target) {
            if (target === 'dist') {
                return grunt.task.run(['build', 'connect:dist:keepalive']);
            }

            grunt.task.run([
                'clean:server',
                'concurrent:server',
                'autoprefixer',
                'connect:livereload',
                'watch'
            ]);
        });

        grunt.registerTask('server', function () {
            grunt.log.warn('The `server` task has been deprecated. Use `grunt serve` to start a server.');
            grunt.task.run(['serve']);
        });

        grunt.registerTask('test', function(target) {
            if (target !== 'watch') {
                grunt.task.run([
                    'clean:server',
                    'concurrent:test',
                    'autoprefixer',
                ]);
            }

            grunt.task.run([
                'connect:test',
                'mocha'
            ]);
        });

        grunt.registerTask('build', [
            'clean:dist',
            'useminPrepare',
            'concurrent:dist',
            'autoprefixer',
            'concat',
            'cssmin',
            'uglify',
            'copy:dist',
            'modernizr',
            'rev',
            'usemin',
            'htmlmin'
        ]);

        grunt.registerTask('default', [
            'newer:jshint',
            'test',
            'build'
        ]);
    };
```

### PR DESCRIPTION
**Pull request by silvenon**
I included it instead of Bootstrap because Bootstrap already ships with Normalize. What do we do in the case when Sass & Compass are also present? Because then both Bootstrap and Normalize aren't included, which leaves `vendor.css` empty again. :worried:

Also, do I add Normalize to the description `Out of the box I include...` or is it unnecessary because Normalize is part of H5BP anyway?

This should fix #242 once it's resolved.

### CODE REVIEW

File: app/templates/index.html [hunk 1]
```diff
@@ -11,7 +11,8 @@
         <!-- Place favicon.ico and apple-touch-icon.png in the root directory -->
         <!-- build:css styles/vendor.css -->
         <!-- bower:css -->
-        <% if (includeBootstrap && !includeCompass) { %><link rel="stylesheet" href="bower_components/bootstrap/dist/css/bootstrap.min.css"><% } %>
+        <% if (includeBootstrap && !includeCompass) { %><link rel="stylesheet" href="bower_components/bootstrap/dist/css/bootstrap.min.css"><% }
+        else if (!includeBootstrap) { %><link rel="stylesheet" href="bower_components/normalize-css/normalize.css" /><% } %>
```
Comment (silvenon) [inline app/templates/index.html]: @eddiemonge how do I write this according to your standards, i.e. closing the tag on the same line where I'm opening it? Is the only solution writing everything on one line?

**Comment by addyosmani** [comment]
I would say we can exclude Normalize from the description.

**Comment by silvenon** [comment]
Bootstrap itself ships with Normalize, so including Normalize again would be redundant and would increase the stylesheet's file size. To clarify:
- no Bootstrap – Normalize should be included
- Bootstrap (no Sass & Compass) – Normalize should **not** be included because Bootstrap already has it, but it's not a problem because `bootstrap.min.css` is included via `bower-install`, `vendor.css` will be built
- Bootstrap + Sass & Compass – Normalize should **not** be included because Bootstrap already has it, but because Bootstrap is included in `main.scss` and not via `bower-install`, `vendor.css` will **not** be built

**Comment by addyosmani** [comment]
Is it possible to drop all references to vendor.css being included at all for the last case?

**Comment by silvenon** [comment]
Yeah, it is. If nobody has a problem with that.

**Comment by silvenon** [comment]
Done. I tested it (manually) and it works correctly in each combination.

**Comment by eddiemonge** [comment]
you are missing bower:css for includebootstrap and compass. bower:css should exist for all options

**Comment by silvenon** [comment]
Ok, but I'm not happy with this solution. I want to include something when Compass and Bootstrap are present. Can't I just include an empty CSS file to avoid the bad request?

**Comment by addyosmani** [comment]
I think we should just include the empty CSS file if it'll avoid any further exceptions being thrown. Does anyone forsee issues with doing that? Would love to finally nail down a solution for this that we can land. It keeps getting reported.

**Comment by silvenon** [comment]
Isn't that an issue with grunt-usemin? Shouldn't it not try to load an asset that doesn't exist? Or should it create an empty one instead?

**Comment by addyosmani** [comment]
I don't think it should try loading an asset that doesn't exist. At the same time, given the slowdown in usemin's release cycle I wonder if we can come to an agreement about a way if fixing this which either involves a workaround factoring in the above discussion or a minor patch to usemin.

**Comment by silvenon** [comment]
A minor patch to usemin would solve this problem more efficiently.

**Pull request closed without merging**